### PR TITLE
Support for multi-FASTA

### PR DESCRIPTION
**Pull request by scatcher125**

### CODE REVIEW

File: hypothesis_bio/hypothesis_bio.py [hunk 1]
```diff
@@ -233,6 +233,7 @@ def fasta(
     sequence_source: SearchStrategy = None,
     wrap_length: Optional[int] = None,
     allow_windows_line_endings=True,
+    num_fasta: Optional[int] = None
```
Comment (mbhall88) [inline hypothesis_bio/hypothesis_bio.py]: Would be better to have `min_entries` and `max_entries` instead. Similar to https://github.com/mbhall88/hypothesis-bio/blob/8d72be5b536bf947938401c313a8af1e727976c2/hypothesis_bio/hypothesis_bio.py#L526

File: hypothesis_bio/hypothesis_bio.py [hunk 2]
```diff
@@ -241,43 +242,67 @@ def fasta(
     - `sequence_source`: The source of the sequence. Defaults to [`dna`](#dna).
     - `wrap_length`: The width to wrap the sequence on. If `None`, mixed sizes are used.
     - `allow_windows_line_endings`: Whether to allow `\\r\\n` in the linebreaks.
+    = `multiple_fasta`: Whether multiple FASTA sequences will be generated.
```
[the remark this answers is not in the document]
Comment (mbhall88) [reply hypothesis_bio/hypothesis_bio.py]: you also need to use `-` instead of `=`

File: hypothesis_bio/hypothesis_bio.py [hunk 3]
```diff
@@ -241,43 +242,67 @@ def fasta(
...
     if sequence_source is None:
         sequence_source = dna()
 
-    comment = draw(comment_source)
-    sequence = draw(sequence_source)
+    if num_fasta is None:
+        # randomly determine how many FASTA sequences to generate
+        num_fasta = draw(integers(min_value=0)) ##replace with numpy if needed
+
+    # Lists to hold comments and sequences for each FASTA sequence
```
Comment (mbhall88) [inline hypothesis_bio/hypothesis_bio.py]: comment not needed

File: hypothesis_bio/hypothesis_bio.py [hunk 4]
```diff
@@ -241,43 +242,67 @@ def fasta(
...
+    # generate pre-determined number of sequences
+    for i in range(num_fasta):
+        comments.append(draw(comment_source))
+        sequences.append(draw(sequence_source))
 
     # the nice case where the user gave the wrap size
     if wrap_length is not None:
-        sequence = fill(sequence, wrap_length, break_on_hyphens=False)
+        for sequence in sequences:
+            sequence = fill(sequence, wrap_length, break_on_hyphens=False)
```
Comment (mbhall88) [inline hypothesis_bio/hypothesis_bio.py]: you don't actually do anything with `sequence` after wrapping it

File: hypothesis_bio/hypothesis_bio.py [hunk 5]
```diff
@@ -241,43 +242,67 @@ def fasta(
...
     # the pathological case
     elif wrap_length is None:
 
-        # choose where to wrap
-        indices = [
-            draw(integers(min_value=0, max_value=len(sequence)))
-            for i in range(draw(integers(min_value=0, max_value=len(sequence))))
-        ]
-        indices = list(set(indices))
+        # list stores lists of indices for each sequences
```
Comment (mbhall88) [inline hypothesis_bio/hypothesis_bio.py]: comment not needed

File: hypothesis_bio/hypothesis_bio.py [hunk 6]
```diff
@@ -241,43 +242,67 @@ def fasta(
...
+        allseq_indices = []
+
+        for sequence in sequences:
+            # choose where to wrap
+            indices = [
+                draw(integers(min_value=0, max_value=len(sequence)))
+                for i in range(draw(integers(min_value=0, max_value=len(sequence))))
+                ]
+            indices = list(set(indices))
```
Comment (mbhall88) [inline hypothesis_bio/hypothesis_bio.py]: Switch the list comprehension to [set comprehension](https://python-reference.readthedocs.io/en/latest/docs/comprehensions/set_comprehension.html) then you don't have to turn a list into a set to turn it back into a list again.

File: hypothesis_bio/hypothesis_bio.py [hunk 7]
```diff
@@ -241,43 +242,67 @@ def fasta(
...
     # sanity checks
     assume("\n\r" not in sequence and "\n\n" not in sequence and "\r\r" not in sequence)
     assume(not sequence.startswith("\r") and not sequence.startswith("\n"))
 
-    return ">" + comment + "\n" + sequence
+    # prepare return string
+    return_str = ""
+    for i in range(len(sequences)):
+        if i!=0:
+            return_str = return_str + "\n"
```
Comment (mbhall88) [inline hypothesis_bio/hypothesis_bio.py]: `return_str += "\n"`

File: hypothesis_bio/hypothesis_bio.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (scatcher125) [reply hypothesis_bio/hypothesis_bio.py]: Thanks for catching that! Updating now

File: hypothesis_bio/hypothesis_bio.py (hunk 1 above)
Comment (scatcher125) [reply hypothesis_bio/hypothesis_bio.py]: Done!

File: hypothesis_bio/hypothesis_bio.py (hunk 3 above)
Comment (scatcher125) [reply hypothesis_bio/hypothesis_bio.py]: Do you mean the comment in the line above or next to num_fasta?

Comment (mbhall88) [reply hypothesis_bio/hypothesis_bio.py]: line 256

File: hypothesis_bio/hypothesis_bio.py [hunk 8]
```diff
@@ -233,6 +233,8 @@ def fasta(
     sequence_source: SearchStrategy = None,
```
Comment (mbhall88) [inline hypothesis_bio/hypothesis_bio.py]: I think `mypy` might also require this to be `Optional[SearchStrategy]`

File: hypothesis_bio/hypothesis_bio.py (hunk 4 above)
Comment (scatcher125) [reply hypothesis_bio/hypothesis_bio.py]: I'm a bit confused about this one. Maybe we can discuss this during the check-in.

**Pull request closed without merging**